### ISSUE TITLE
why is the mpu/cross_entropy missing a softmax_logits_t

### ISSUE THREAD

**Issue by 155394551lzk** [microsoft/LMOps#157]
In LMOps/minillm/transformers/src/transformers/mpu /cross_entropy.py/_ParallelSoftCrossEntropyLoss line144
the ceLoss = torch.log(sum_exp_logits) - sum_targets_softmax_logits 
but the whole ceLoss should = softmax_logits_target * torch.log(sum_exp_logits) - sum_targets_softmax_logits.

Thanks!
![Q0FXCFUGQF1TWFhXB0xYVV9DBQgBBgYPAwMHATY2Mzk+](https://github.com/microsoft/LMOps/assets/60507390/e217f96e-955a-4632-a41c-93e85de4789f)

**Comment by t1101675** [comment]
The CE loss should sum over the vocabulary:
<img width="665" alt="image" src="https://github.com/microsoft/LMOps/assets/38183190/064de1af-4da7-4d34-9ff5-c2de582374b7">

**Comment by 155394551lzk** [comment]
get it!
